homeassistant/components/tahoma.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support for Tahoma devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/tahoma/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_EXCLUDE
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import (slugify)
+
+REQUIREMENTS = ['tahoma-api==0.0.10']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'tahoma'
+
+TAHOMA_ID_LIST_SCHEMA = vol.Schema([cv.string])
+
+TAHOMA_ID_FORMAT = '{}_{}'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_EXCLUDE, default=[]): TAHOMA_ID_LIST_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Activate Tahoma component."""
+    from tahoma_api import TahomaApi
+
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    exclude = conf.get(CONF_EXCLUDE)
+    try:
+        api = TahomaApi(username, password)
+        hass.data['TAHOMA_CONTROLLER'] = api
+    except HomeAssistantError:
+        _LOGGER.exception("Error communicating with Tahoma API")
+        return False
+
+    try:
+        api.get_setup()
+        devices = api.get_devices()
+    except HomeAssistantError:
+        _LOGGER.exception("Cannot fetch informations from Tahoma API")
+        return False
+
+    hass.data['tahomacover'] = []
+    hass.data['tahomasensor'] = []
+
+    for device in devices:
+        _device = api.get_device(device)
+        if any(ext not in _device.type for ext in exclude):
+            device_type = map_tahoma_device(_device)
+            if device_type is None:
+                continue
+            hass.data[device_type].append(_device)
```

Comment:
Use platform descovery to discover your platforms. You can also send config with that, so you don't need store it into `data`. Look into https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/tesla.py or https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/android_ip_webcam.py